api/v1beta1/properties.go

```diff
@@ -108,22 +108,22 @@ func (dk *DynaKube) ActiveGateImage() string {
 	return fmt.Sprintf("%s/linux/activegate:latest", registry)
 }
 
-func (dk *DynaKube) ServerlessMode() bool {
+func (dk *DynaKube) UseCSIDriver() bool {
```

Comment:
Can we combine `UseCSIDriver()` and `NeedsCSI()` into a single function that is used to check if csi driver should be enabled?

Only the result of all the checks (`NeedsCSI()` currently) should be exposed and used in other parts of the code.